File: src/AZMongoBak/ClientApp/src/data-models/mini-backup-info.ts
import { APIService } from "../auth/api-service";

export class MiniBackupInfo {

    public oid = "";
    public display_name = "";

    public static async ListAsync(api: APIService): Promise<MiniBackupInfo[]> {

        let infos: MiniBackupInfo[] = [];

        try {
            if (await api.InitAsync()) {
                const header = api.GetHeaders();
                const url = `${api.BASE}/backupInfo/list-mini`;
                const request = await fetch(url, { headers: header });

				if (request.ok) {
                    infos = await request.json() as MiniBackupInfo[];
                }
            }
        }
        catch (err) {
            console.error("MiniBackupInfo.ListAsync", err);
        }

        return infos;
    }
}

File: src/AZMongoBak/Models/MiniBackupInfo.cs
using AZMongoBak.MongoDb.Collections;

namespace AZMongoBak.Models {
    public class MiniBackupInfo {
        public string oid { get; set; } = "";
        public string display_name { get; set; } = "";

        public MiniBackupInfo() {}
        public MiniBackupInfo(BackupInfo info) {
            this.oid = info.oid;
            this.display_name = info.display_name;
        }
    }
}